folium/features.py

```diff
@@ -241,8 +241,19 @@ def __init__(self, data, style_function=None):
             else:  # This is a filename
                 self.embed = False
                 self.data = data
+        elif data.__class__.__name__ in ['GeoDataFrame','GeoSeries']:
+            if hasattr(data, '__geo_interface__'):
+                # We have a GeoPandas 0.2 object
+                self.embed = True
+                self.data = json.loads(json.dumps(data.to_crs(epsg='4326').__geo_interface__))
+            elif hasattr(data, 'columns'):
+                # We have a GeoDataFrame 0.1
+                self.embed = True
+                self.data = json.loads(data.to_crs(epsg='4326').to_json())
+            else:
+                raise ValueError('Unable to transform this object to a GeoJSON.')
         else:
-            raise ValueError('Unhandled data type.')
+            raise ValueError('Unhandled data type {!r}.'.format(data))
```

Comment:
Let's change the `data type` to `object`, since we are printing the `__repr__` and not the `type`.